src/cpu.rs

```diff
@@ -527,6 +549,181 @@ impl Cpu {
         self.condition_codes.set_zero(self.registers.a);
         self.condition_codes.set_parity(self.registers.a);
     }
+
+    // arithmetic group
+    fn add(&mut self, val: u8) {
+        // val is value containted in register
+        let reg_a = self.registers.a;
+        let res: u16 = (reg_a as u16).wrapping_add(val as u16);
+        // put result in accumulator
+        self.registers.a = res as u8;
+        // update flags
+        self.condition_codes.set_zero(res as u8);
+        self.condition_codes.set_sign(res as u8);
+        self.condition_codes.set_parity(res as u8);
+        self.condition_codes.set_carry(res > 0xFF);
+        self.condition_codes
+            .set_aux_carry((reg_a & 0x0F) + (val & 0x0F) > 0x0F);
+    }
+
+    fn inr(&mut self, reg: Operand) {
```

Comment:
I think I see what you were asking about yesterday. I'm assuming you were trying to pass a reference to the register value here to get the macro to work? I think this is fine as is without that work. Can always change that later if we need or want to.

Reply:
Yeah, that is what I was trying to do, I can write it that way, I got it mostly figured out, but it seemed silly to write a macro for two functions.